src/components/profile.ts

```diff
@@ -288,10 +232,41 @@ const updateMemberGradRoles = async (member: GuildMember, gradYear: number): Pro
     let roleToRemove = oldYear.toString();
     if (oldYear < validRoleYears.validRoleYearStart) {
       // while profile year may be something like 2008, the role itself is called 2000s so have to convert
-      roleToRemove = yearToDecade(oldYear).toString() + 's';
+      roleToRemove = yearToDecade(oldYear) + 's';
     }
     if (newYearRoleName !== roleToRemove) {
       updateMemberRole(member, roleToRemove, addOrRemove.remove);
     }
   }
 };
+
+export const assignDecadeAndPruneYearRoles = async (): Promise<void> => {
+  const guild = await container.client.guilds.fetch(TARGET_GUILD_ID);
+  guild.roles.cache
+    .filter((role) => {
+      const roleNameNum = Number(role.name);
+      return Number.isInteger(roleNameNum) && roleNameNum < validRoleYears.validRoleYearStart;
+    })
+    .forEach((role) => {
+      const roleMembers = role.members;
+      const decadeRoleName = yearToDecade(Number(role.name)) + 's';
+      roleMembers.forEach((member) => updateMemberRole(member, decadeRoleName, addOrRemove.add));
+      guild.roles.delete(role);
+    });
+};
+
+export const assignAlumniRole = async (): Promise<void> => {
+  (await container.client.guilds.fetch(TARGET_GUILD_ID)).roles.cache
```

Comment:
We might just want to query `roles` directly, since the cache might not contain the roles. i.e. `roles.filter. ...` rather than `roles.cache.filter ...`

Reply:
Alright, I fetched the roles.